envi/archs/ppc/vle.py

```diff
@@ -389,6 +390,39 @@ def case_E_XRA(types, data, va):
     INS_ADD: 4, # not used but needs to be present since it's E_D
 }
 
+
+# Simplified mnemonic transformation functions
+def simpleE_ORI(ival, mnem, opcode, opers, iflags):
+    if len(opers) == 3 and \
```

Comment:
This may be called often. In this regard simple optimizations here will pay off. The way this code is written len(), isReg() and isImmed() may be called twice instead of once. Python has no way to optimize this at compile time yet. Consider moving the len(opers) == 3 up and then do the rest of the checks. Even better would be to call isReg and isImmed once and assign to a variable only then do logic checks.

Reply:
thanks, @bat-serjo , good call.
alternately, since both major conditional sequences have the same set, we could break this into a nested if:

```
if len(opers) == 3 and \
        opers[0].isReg() and \
        opers[1].isReg() and \
        opers[2].isImmed() and \
        opers[2].val == 0 and \
        iflags == IFLAGS_NONE:
    if opers[0].reg == 0 and opers[1].reg = 0:
        return 'e_nop', INS_NOP, tuple(), iflags
    elif opers[0].reg == opers[1].reg:
        return 'e_mr', INS_MR, opers[:2], iflags

return mnem, INS_ORI, opers, iflags
```

Reply:
Thank you for noticing that, I'll go fix this.